### PR TITLE
[FrameworkBundle] Add missing items in the unused tag pass whitelist

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | n/a 
| License       | MIT
| Doc PR        | n/a

We have some missing tags in the whitelist. I've added a script that adds the missing ones, and added a test to avoid forgetting about updating the whitelist.

### CODE REVIEW

**Review by chalasr — APPROVED** [review]
Nice one

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/UnusedTagsPass.php [hunk 1]
```diff
@@ -42,12 +49,20 @@ class UnusedTagsPass implements CompilerPassInterface
...
+        'kernel.reset',
+        'mailer.transport_factory',
         'messenger.bus',
-        'messenger.receiver',
         'messenger.message_handler',
+        'messenger.receiver',
+        'messenger.transport_factory',
         'mime.mime_type_guesser',
         'monolog.logger',
-        'proxy',
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/UnusedTagsPass.php]: Just asking: is it ok to be removed from the list ?

Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/UnusedTagsPass.php]: fixed

**Review by yceruto — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/UnusedTagsPassTest.php [hunk 2]
```diff
@@ -31,4 +31,23 @@ public function testProcess()
 
         $this->assertSame([sprintf('%s: Tag "kenrel.event_subscriber" was defined on service(s) "foo", "bar", but was never used. Did you mean "kernel.event_subscriber"?', UnusedTagsPass::class)], $container->getCompiler()->getLog());
     }
+
+    public function testMissingWhitelistTags()
+    {
+        $this->assertSame(UnusedTagsPassUtils::getDefinedTags(), $this->getWhitelistTags(), 'The src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/UnusedTagsPass.php file must be updated; run src/Symfony/Bundle/FrameworkBundle/Resources/bin/check-unused-tags-whitelist.php.');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/UnusedTagsPassTest.php]: We should skip this test for deps=* tests. I'd suggest doing so this way:
`if (!defined('PHPUNIT_COMPOSER_INSTALL') || dirname(PHPUNIT_COMPOSER_INSTALL, 2) !== dirname(__DIR__, 5)) // skip`
(if I counted correctly for the 2nd args)

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/UnusedTagsPassTest.php]: hum this check doesn't work :)
Another blind try:
`dirname((new ReflectionClass(CompilerPassInterface::class))->getFileName(), 3) === dirname(__DIR__, 4)`

**Comment by nicolas-grekas** [comment]
Thank you @fabpot.

**Pull request merged**